### ISSUE TITLE
Mumble support

### ISSUE THREAD

**Issue by grappas** [IDI-Systems/acre2#577]
Since Teamspeak producers make even more restrictions lately for getting non-profit license, It would be good idea to create Mumble plugin. Are you planning?

**Comment by Soldia1138** [comment]
Yes

To be more precise: we are evaluating. Mumble Development seems to be stuck as well.

**Comment by grappas** [comment]
Sounds great 👍

**Comment by Krzmbrzl** [comment]
I'm currently trying to push the Mumble plugin support forward - see https://github.com/mumble-voip/mumble/issues/3742
So if you want to have certain features in there, make sure to leave a comment in the linked issue.

For this I am looking into writing the plugin system (partly) myself. I'll do it in a WIP fashion so I will create PRs at very early stages so that interested people (maybe some of you guys) can start criticizing my code early on. (e.g. https://github.com/mumble-voip/mumble/pull/3743)

**Comment by jonpas** [comment]
@Krzmbrzl We are following and will post our requirements soon (@NouberNou).

Thank you for your efforts!

**Comment by hbeni** [comment]
> To be more precise: we are evaluating.

@jonpas @NouberNou I stumbled upon this ticket by chance, and wanted to give you a short notice that i already developed a plugin for a similar system for flight simulators: [hbeni/fgcom-mumble](https://github.com/hbeni/fgcom-mumble).
Maybe you can take some pieces from there, but a radio system is entirely doable using the new API.

**Comment by jokoho48** [comment]
@hbeni the ACRE Devs are already working on an Implementation actively: https://github.com/IDI-Systems/acre2/pull/980

**Comment by hbeni** [comment]
Ah, cool to hear! Thanks!

**Comment by TheMagnetar** [comment]
Yeah. We started development in early April. It is definitely nice to see it also being adopted in flight sims. Flight gear has a neat implementation of Longley-Rice!

**Comment by Krzmbrzl** [comment]
The referenced pull request (mumble-voip/mumble#3743) has now been merged.
This means that plugin support is now available in all future Mumble feature releases (1.4.0 onwards) - including snapshot releases.

**Comment by Krzmbrzl** [comment]
@PabstMirror @mharis001  I have been told on Slack that you guys are now the main devs that (are supposed to) continue work on ACRE2.
As such I wanted to ask what your plans in terms of the Mumble plugin are and if there is something I can help to get it up-to-date.

Given that this has been working before (with an in-progress version of the Mumble API), I am kinda expecting that the main work that has to be done on it is to adapt the used API.
I could probably do that for you, but I would like to get help on the actual ACRE stuff if there is something that goes beyond the mere Mumble API. Do you think that would be possible? Or do you currently have too much other stuff going on? :eyes:

**Comment by Krzmbrzl** [comment]
@jonpas Do you perhaps have an insight into what the current plans are regarding the Mumble plugin?
(Pinging you as I think you might have a clue :D)

**Comment by jonpas** [comment]
I can tell you that none of the [Managers](https://acre2.idi-systems.com/team#managers) have time to do this, we are super busy with other things at the moment.

@PabstMirror or @mharis001 might pick it up, it is entirely up to them, I am not sure if any of them decided to just yet. Any contributor is also welcome to pick it up and update it for the latest API of course!

Hopefully that answers your question.

**Comment by Krzmbrzl** [comment]
Alright. Then I'll try to find some time to update the plugin to the latest API specs and create a PR for that. This will only work on the core Mumble API level though as I don't have a clue of ACRE internals xD

I assume the PR should be created against the `mumble-plugin` branch?

**Comment by jonpas** [comment]
Yeah, PR to that branch would be perfect.

**Comment by Krzmbrzl** [comment]
See https://github.com/IDI-Systems/acre2/pull/1108 (WIP)

**Comment by mwpowellhtx** [comment]
Interesting.

Does _Mumble_ operate as what? An application service analogous to _TeamSpeak_?

Working on an _Always Up_ alternative to help with the (self-)hosting side of things.

**Comment by Krzmbrzl** [comment]
Not sure what you are asking but Mumble is essentially the same kind of application as TeamSpeak (aka: a voice communication software).
For more info, check https://www.mumble.info/

**Comment by mwpowellhtx** [comment]
> Not sure what you are asking but Mumble is essentially the same kind of application as TeamSpeak (aka: a voice communication software). For more info, check https://www.mumble.info/

Yes, I understand that, with some mixed reviews re: sound quality, administrative proficiency, etc.

How feature API stable is it today? Given the PR was some 3+ years ago now. My assumption is, probably quite stable.

How good is it at self-hosting?

With TS, for instance, I have learned _Always Up_ is a popular, but trial freemium, solution, allowing it to operate as an application service.

Anyway, I am working on an open source AU alternative.

**Comment by hbeni** [comment]
@mwpowellhtx 
The mumble server is easy to install in linux and can start with the system boot, so that is probably a far better/easier solution to fiddling around with some windows beef things.

My experience hosting a mumble server on debian linux is good for the last three years.

**Comment by mwpowellhtx** [comment]
> @mwpowellhtx The mumble server is easy to install in linux and can start with the system boot, so that is probably a far better/easier solution to fiddling around with some windows beef things.
> 
> My experience hosting a mumble server on debian linux is good for the last three years.

Well... although I have WSL2, I'm not sure I want to install it there. So Windows Service Application is the alternative method.

**Comment by Krzmbrzl** [comment]
This is OT for this issue here so this will be my last answer regarding this topic. If you want to continue the discussion, I encourage you to join the Mumble Matrix channel.

> How feature API stable is it today? Given the PR was some 3+ years ago now. My assumption is, probably quite stable.

Completely stable

> How good is it at self-hosting?

That's like the primary use case, so I would say excellent. 

> With TS, for instance, I have learned Always Up is a popular, but trial freemium, solution, allowing it to operate as an application service.

No idea what _Always Up_ is. Mumble itself is not a Windows service and I have no clue on how these work and whether or not Mumble can be made to be one.

**Comment by mwpowellhtx** [comment]
Appreciate the feedback. Cheers 🍻

**Comment by demif1end** [comment]
(None of this is a demand, I recognize there are better things to do right now)
I would like to highlight possibly renewing interest in the concept of bringing this to Mumble; TeamSpeak 3 is scheduled for End-of-Life in December of 2027, and it is essentially inevitable by that point with how old TS3 is getting as a platform and, bluntly, TeamSpeak's behavior in the past that we have no idea what's going to happen after that. In the best case, they open source it and let the community handle it; in the middling case, they make getting a hold of the software, or the NPL, even harder; in the worst case, we get a billion CVEs and the server is a liability to have on your rackspace, so it may be a good idea to reassess the priority. A much less pressing reason is, of course, Mumble is open-source with a very permissive license and you can do whatever you want without paying any corporation a dime (or anyone at all, unless you wanna [sponsor them](https://liberapay.com/mumble), which you should if you can, smile).

I'm not even averse to working on this myself, I just know little to nothing about how this works internally, so I don't _know_ if I know enough to assist directly.

December 2027 is quite a ways away, but (IN MY PERSONAL OPINION) it's probably a good idea to get ahead of at least thinking about it. The prospective transition for existing users (and server admins) might be a little agonizing if it's hard and fast, and it'd soften it a little.

Cheers to everyone working on ACRE2, this mod rocks and I wanna see it live forever.

**Comment by mwpowellhtx** [comment]
> I would like to highlight possibly renewing interest in the concept of bringing this to Mumble; TeamSpeak 3 is scheduled for End-of-Life in December of 2027, and it is essentially inevitable by that point with how old TS3 is getting as a platform and, bluntly, TeamSpeak's behavior in the past that we have no idea what's going to happen after that.

Business models, licensing, and IP issues aside, indeed...

> In the best case, they open source it and let the community handle it; in the middling case, they make getting a hold of the software, or the NPL, even harder; in the worst case, we get a billion CVEs and the server is a liability to have on your rackspace, so it may be a good idea to reassess the priority.

> A much less pressing reason is, of course, Mumble is open-source with a very permissive license and you can do whatever you want without paying any corporation a dime (or anyone at all, unless you wanna [sponsor them](https://liberapay.com/mumble), which you should if you can, smile).

The main issue is, when a TS3 and it's protocols, integration, expires, does that also mean similarly based platforms do, i.e. Mumble. i.e. what is the migration path.

> I'm not even averse to working on this myself, I just know little to nothing about how this works internally, so I don't know if I know enough to assist directly.

It'd be curious for me, with some help getting there. I have other irons in the fire, this would be a curious thing for my part.

> December 2027 is quite a ways away, but (IN MY PERSONAL OPINION) it's probably a good idea to get ahead of at least thinking about it. The prospective transition for existing users (and server admins) might be a little agonizing if it's hard and fast, and it'd soften it a little.

As I've stated, been a minute since I've done much of anything A3 wise. It's a bit of an awkward timing, IMHO, not least of which horrible timing given the downturns in world wide economic prosperity. No signs of getting much better any time soon. Vis a vis A3, BIS in general, for them especially awkward, somewhere between A3 and A4, with the in between Reforger.

> Cheers to everyone working on ACRE2, this mod rocks and I wanna see it live forever.

There is no forever, not in today's marketplace. The only question is, as you alluded to, rightly so, is plausible migration path.

The question in my mind is what this all means in the broader context, sunset on A3, mods, etc, for instance. Migration through Reforger to A4, perhaps, similar comparable modding, along these lines.

Just some mud on the wall thoughts in response FWIW. Cheers and best.

**Comment by mwpowellhtx** [comment]
@demif1end FWIW, I asked this prompt to deepseek and the response was insightful.

> consider Bohemia Studios Arma 3, concerning mods supporting realistic radio simulation, involving TeamSpeak 3 and Mumble API. given TeamSpeak end of life is coming in 2027, and taking into account also Arma 3 end of life, what is the communications migration path through Reforger into Arma 4, involving such communications, radios, and similar community modding.

The highlights in response FWIW...

> The Current State (Arma 3) – Why it's Fragile, TFAR, ACRE2, "the problem"...
> Reforger – The "Proof of Concept" Migration, No external API reliance... Modding in C# and Enforce Script... The Current Path for Reforger...
> Arma 4 – The Final Destination (Native-Centric Architecture), Deprecation of TS3/Mumble as primary drivers... The new pipeline... External VoIP becomes optional, not mandatory...
> The Community Modding Migration Roadmap, 2025–2026 (Transition Phase)... 2026–2027 (Reforger Native Ports)... 2027+ (Arma 4 Launch)
> What About TeamSpeak 5 and Discord? TeamSpeak 5... modern WebSocket based SDK, not game-injection DLL... Modders will build standalone "Radio Relay" desktop apps that connect to Arma 4 via the official Enfusion Remote Exec interface...
> The Bottom Line for Your Community... Do not panic about 2027. Your real migration starts with Reforger... The golden rule moving forward...

The golden rule being of course, "server dominance" the server dictates how to engage with the community. This has always been the case, which mods are expected, allowable, etc, etc.

Arma 4 end game is especially insightful, "Radio mods will be entirely self-contained within the Enfusion ecosystem."

> In short: the migration path is away from TS3/Mumble plugins and toward Enfusion's native audio pipeline, with external apps relegated to high-fidelity add-ons via modern web protocols—not legacy C++ hooks. The modding community is already 2 years into this rewrite, and Arma 4 will launch with radio simulation that is more immersive, more stable, and entirely independent of third-party EOL schedules.



YMMV...

Good luck!

**Comment by demif1end** [comment]
> The main issue is, when a TS3 and it's protocols, integration, expires, does that also mean similarly based platforms do, i.e. Mumble. i.e. what is the migration path.

Mumble has nothing to do with TeamSpeak 3. It is an entirely separate project with a different protocol, team, etc. It will not die with TeamSpeak 3.

> There is no forever, not in today's marketplace.

Dreams are permitted from time to time.

I think considering Arma 4 at all here is a little out of scope. I would expect Arma 4 to sidestep the need for ACRE and it's descendents entirely.
Also, Arma 3 can't really go.. end of life? I mean, yes, it can, they can stop providing content updates and whatnot and shift development focus, but the game runs it's own master servers, has a very well-living workshop, and is one of the largest community platforms for this sort of thing ever if not the largest. Arma 4 will launch and people, by and large will still be sticking with Arma 3 primarily who already play it for, at the absolute least, 5 years, solely because of the wealth of custom content and modularity Arma 3 already offers. The mission statement proposed here is to help ACRE2 last for that period, however long it may be, and then hopefully some more. Bohemia doesn't really have a tendency to make things inaccessible. I would expect Arma 3 to stay perfectly preserved and maintained for quite a while; they realize exactly how important it is.

TeamSpeak 5/6 is impractical, in my opinion; they work in a way that will make people upset coming from the simplicity of pure VOIP/text applications like the prior TS3. Discord is even more impractical, fundamentally not supporting something like this conceptually.

Mumble and TeamSpeak 3 have a lot in common, except one's free and is significantly less likely to go anywhere. Both Mumble and TeamSpeak 3 work on the same concepts and even at times share similar APIs to the point that Mumble could be considered an "OpenTeamSpeak" of sorts.

Mumble is also very well-suited to this project specifically; it has the lowest latency and highest quality on average out of any common VOIP application while simultaneously being encrypted just as TeamSpeak 3 is, and has similar 3D positional audio APIs, and a functioning plugin system, etc, etc. Mumble, from my view, is the path forward for *ACRE2*. ACRE2 works well. There is little reason to replace ACRE2 for Arma 3, Arma 4 probably won't need it, and I'm not looking to propose some startup-level "visionary" concept of The Future. The path forward, for what ACRE2 is, and what it does, is Mumble, and the only reason it's Mumble is cause TeamSpeak 3 is on it's way out. This is not only beneficial for the longevity of the mod, but for server admins, it's significantly less of a hassle. Mumble works roughly the same from a user and server administrator perspective as TeamSpeak 3. Everything works how everyone remembers it, for the most part, except now it's not bleeding out, and you can have a 128 slot server without giving anyone cash. Yay!

Relying on a statistical probability machine with injected randomness for insight is not my cup of tea, but to each their own. Fair travels.

**Comment by hbeni** [comment]
(Sorry for the noise, didnt remember I already mentioned fgcom-mumble in 2020)
